velox/type/OpaqueCustomTypes.h

```diff
@@ -55,9 +55,9 @@ class OpaqueCustomTypeRegister {
    public:
     VeloxType() : OpaqueType(std::type_index(typeid(T))) {}
 
-    static const TypePtr& get() {
-      static const TypePtr instance{new VeloxType()};
-      return instance;
+    static TypePtr get() {
+      static const VeloxType kInstance;
+      return {TypePtr{}, &kInstance};
```

Comment:
@MBkkt I'm seeing errors internally. I believe they are caused by this change.

```
error: non-const lvalue reference to type 'shared_ptr<...>' cannot bind to a temporary of type 'shared_ptr<...>'
   26 |   auto& typePtr = T::singletonTypePtr();
      |         ^         ~~~~~~~~~~~~~~~~~~~~~
```

Reply:
I do not think it is feasible to change the callsites. Would you revert this specific change?